src/ir/mod.rs

```diff
@@ -609,6 +613,21 @@ impl Substitution {
     pub fn is_empty(&self) -> bool {
         self.tys.is_empty() && self.lifetimes.is_empty()
     }
+
+    pub fn into_parameters(self) -> Vec<Parameter> {
```

Comment:
Would appreciate a review of this function @aturon. In particular it assumes:

* The index of the inference variables is the same index we started with in the query.
* Every variable will have a substitution, so there won't be any gaps.

I don't know if this is true or not but if it isn't I don't see how we could transform this to the subst folder, which takes a slice of parameters and uses their indexes.